src/main/java/core/basesyntax/service/impl/FileReaderImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements core.basesyntax.service.FileReader {
+    @Override
+    public List<String> readFromFile(String fromFile) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFile))) {
+            List<String> inputList = new ArrayList<>();
+            String value = reader.readLine();
+            while (value != null) {
+                inputList.add(value);
+            }

```

Comment:
There is an infinite loop in the while statement, as `value` is not being updated within the loop. You need to read the next line inside the loop to update the `value` variable.